src/core/ext/filters/client_channel/health/health_check_client.cc

```diff
@@ -21,13 +21,14 @@
 #include <stdint.h>
 #include <stdio.h>
 
+#include <grpcpp/impl/codegen/sync.h>
```

Comment:
I don't think it's okay for code in core to depend on headers in the grpcpp directory.  We might need to duplicate this code in src/core/lib/gprpp.

Reply:
So @soheilhy and I were discussing this, and this is an interesting point. This looks unkosher, but it's no more unkosher than using std::unique_ptr as this is a headers-only library. There is no linkage involved.

Reply:
Yes, that's a great point. @vjpai and I discussed this offline before I created the patch.

The idea that I'm proposing is that , instead of duplicating the code, we can allow usage of header from "<grpcpp/...>" only when (a) they are header-only and (b) have no run-time stdlib dependency.

Could you please let me know your thoughts? Does this sound reasonable to you?

Reply:
Oh really sorry, cross-posted with @vjpai.

Reply:
Well, at least we both confirmed that we agree on this.

Reply:
Good point.

Would it make sense to have a header in src/core/lib/gprpp that does nothing but include the header from include/gprpp and provide a comment saying why that's okay, and then use that header in core instead of directly including from include/gprpp?  That would document this for future readers of the code.

Reply:
That's a great idea. We can just include them there. We can possibly just alias them in `grpc_core::` so that we don't have to use `grpc::internal::`. Would you guys have any objection to aliasing?

Reply:
That's a good idea -- aliasing sounds great!

Reply:
Great, will do then (hopefully by tomorrow afternoon, as something urgent came up this morning).

Reply:
Aliasing sounds great to me too. Can we also request input from @nicolasnoble so

Reply:
Sure added @nicolasnoble to the reviewers.